app/controllers/recipes_controller.rb

```diff
@@ -18,4 +20,25 @@ def show
     @recipe = Recipe.find(params[:id])
     @current_user_bookmarks = Bookmark.where(user: current_user)
   end
+
+  private
+
+  def user_ingredients(array)
+    array.map! { |ingredient_id| Ingredient.find(ingredient_id).name }.join(" ")
+    if params.values.include?("1")
+      checked = params.select { |_key, value| value == "1" }.keys
+      @recipes = Recipe.search_recipes("#{params[:query]} #{str}").where(checked.to_h { |value| [value, true] })
```

Comment:
Just wondering, where does the interpolates str variable come from. Is it defined elsewhere?

Reply:
Thanks for flagging - I didn't update that! Will sort it when I'm home and refactor a little as it needs work..